_core/smart/tutorial-counter.md

```diff
@@ -0,0 +1,193 @@
+        ✓ should have a valid syntax
+        deploying an instance of the contract
+        1) should start at zero
+        2) should increment
+        3) should decrement
+
+    hello world contract test suite
+        ✓ should have a valid syntax
+        deploying an instance of the contract
+        ✓ should return 'hello world'
+        ✓ should echo number
+
+
+    4 passing (455ms)
+    3 failing
+    ```
+
+    It looks like we see some failed test! That is on purpose - we will implement the new smart contract in the next steps! After every increment of the contract, we will rerun the tests to ensure we're on the right track.
+
+## Step 3: Developing a smart contract
+
+Let's get familiar with the tests to understand what the new smart contract should look like
+
+1. Take a quick look at the test file associated with the counter smart contract:
+
+    ```shell
+    cat test/counter.ts
+    ```
+
+    You should be familiar with the test set up from the Hello World tutorial. Notice how the instance of the smart contract is created on line 8:
+
+    ```js
+    counterClient = new Client("SP3GWX3NE58KXHESRYE4DYQ1S31PQJTCRXB3PE9SB.counter", "counter", provider);
+    ```
+
+    That tells us that the new smart contract is named `counter` and that we need to start by creating a new file for the smart contract: `contracts/counter.clar`. Note that the `contracts` folder is assumed as the base folder and that every Clarity file has the suffix `.clar`.
+
+2. Let's create the new file:
+
+    ```shell
+    touch contracts/counter.clar
+    ```
+
+3. With the editor of your choice, open the file and add the following lines of code:
+
+    ```cl
+    (define-data-var counter int 0)
+
+    (define-public (get-counter)
+        (ok (var-get counter))
+    )
+    ```
+
+    The first line initializes a new integer variable `counter` with the value set to `0` using the [`define-data-var`](https://docs.blockstack.org/core/smart/clarityref#define-data-var) statement. It is important to note that all definition statements in Clarity need to be at the top of the file.
+
+    To provide access to the variable from outside of the current smart contract, we need to declare a public get method for this variable. The last lines of the code add a public `get-counter` method. The [`var-get`](https://docs.blockstack.org/core/smart/clarityref#var-get) statement looks for a variable in the contract's data map and returns it.
+
+    With that, you are ready to rerun the tests!
+
+4. Run the tests and review the results:
+
+    ```shell
+    npm test
+    ```
+
+    You should now only see 2 failing tests! `should start at zero` is passing, and you successfully build your first part of the contract. Congrats!
+
+    However, we don't stop here. Let's implement increment and decrement methods.
+
+5. Add the following lines to the bottom of the `counter.clar` file and take a few seconds to review them:
+
+    ```cl
+    (define-public (increment)
```

Comment:
Again, we don't have formal style guides yet, but most/all of the Clarity code I've seen so far uses 2-space indentation. It's also what we use in the hello-world starter. Would recommend switching to 2 spaces

Reply:
fixed